radix_sort.py

```diff
@@ -0,0 +1,78 @@
+from Queue import Queue
+import random
+import timeit
+
+
+def radix_sort(a_list):
+    queue_list = [Queue() for i in range(10)]
+    maximum = 0
+    for item in a_list:
+        if item > maximum:
+            maximum = item
+        queue_list[item % 10].put(item)
+    a_list_pointer = 0
+    for each_queue in queue_list:
+        while not each_queue.empty():
+            a_list[a_list_pointer] = each_queue.get()
+            a_list_pointer += 1
+
+    max_digits = len(str(maximum))
+    _radix_sort(a_list, max_digits, queue_list)
+
+
+def _radix_sort(a_list, max_digits, queue_list):
+    for i in xrange(1, max_digits):
+        for item in a_list:
+            queue_list[(item // 10**i) % 10].put(item)
+        a_list_pointer = 0
+        for each_queue in queue_list:
+            while not each_queue.empty():
+                a_list[a_list_pointer] = each_queue.get()
+                a_list_pointer += 1
```

Comment:
I suspect that if you worked on it, you could get rid of this helper method, and deal with the sort merely as a single iteration for powers of 10 in `range(max_digits)`